src/ecalc/libraries/libecalc/common/libecalc/dto/components.py

```diff
@@ -467,14 +469,28 @@ def validate_unique_names(cls, values):
                 if isinstance(fuel_consumer, GeneratorSet):
                     for electricity_consumer in fuel_consumer.consumers:
                         names.append(electricity_consumer.name)
+                if fuel_consumer.fuel is not None:
+                    for fuel_type in fuel_consumer.fuel.values():
+                        # Need to verify that it is a different fuel
+                        if fuel_type is not None and fuel_type not in fuel_types:
+                            fuel_types.append(fuel_type)
+                            fuel_names.append(fuel_type.name)
 
         duplicated_names = get_duplicates(names)
+        duplicated_fuel_names = get_duplicates(fuel_names)
```

Comment:
This check is actually doing that, checking if it is a different fuel. The fuel_types list includes the whole fuel, not only the name. So, this will catch situations where e.g. 2 fuels are called fuel 1 for co2 but some other values are different:

https://github.com/equinor/ecalc/blob/c41166fc458636aa35748a9438fa2db4e5bdca81/src/ecalc/libraries/libecalc/common/libecalc/dto/components.py#L475

As far as I understand, the only thing we don´t catch is the situation where the user has duplicated the fuel and forgets to change parameters.

Reply:
I have added a check for duplicate fuels in yaml. This should capture the copy-paste issue where the user forgets to change name etc.:

https://github.com/equinor/ecalc/blob/5fceec63355229a9179ec740e12184f87e83a986/src/ecalc/libraries/libecalc/common/libecalc/input/mappers/create_references.py#LL31C1-L31C1

Also added a test with yaml file containing duplicate fuels:
https://github.com/equinor/ecalc/blob/5fceec63355229a9179ec740e12184f87e83a986/src/ecalc/cli/tests/test_app.py#L683